### PR TITLE
Split Startup into Program.cs and Startup.cs

### PR DESCRIPTION
**Pull request by phenning**
@rustd @davidfowl 

Same as the sample, with the exception of the added core 4 template namespaces.

### CODE REVIEW

**Comment by phenning** [comment]
@DamianEdwards

File: src/BaseTemplates/EmptyWeb/Program.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public class Program
+    {
+        // Entry point for the application.
+        public static void Main(string[] args)
+        {
+           var configuration = WebApplicationConfiguration.GetDefault(args);
+
+            var application = new WebApplicationBuilder()
+                        .UseApplicationBasePath(Directory.GetCurrentDirectory())
```
Comment (DamianEdwards) [inline src/BaseTemplates/EmptyWeb/Program.cs]: Just a single indent level here.

File: src/BaseTemplates/EmptyWeb/Program.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+namespace $safeprojectname$
+{
+    public class Program
+    {
+        // Entry point for the application.
+        public static void Main(string[] args)
+        {
+           var configuration = WebApplicationConfiguration.GetDefault(args);
+
+            var application = new WebApplicationBuilder()
```
Comment (DamianEdwards) [inline src/BaseTemplates/EmptyWeb/Program.cs:17]: `var hostingConfiguration = WebApplicationConfiguration.GetDefault(args)`

Comment (phenning) [inline src/BaseTemplates/EmptyWeb/Program.cs:17]: Updated. Also fixed the indent level on line 15, it was off by one space.

File: src/BaseTemplates/EmptyWeb/Program.cs (hunk 1 above)
Comment (phenning) [inline src/BaseTemplates/EmptyWeb/Program.cs]: Update this as well

**Comment by davidfowl** [comment]
Need to figure out the app base thing. It's currently broken on IIS using the current directory.

**Comment by phenning** [comment]
@davidfowl  Should we just do the split for now and level the ApplicationBasePath out of this PR?

**Comment by phenning** [comment]
Talked to fowler in person, we're ok with the appbase bits, submitting

**Pull request closed without merging**